cpp/src/io/orc/stripe_data.cu

```diff
@@ -655,8 +655,12 @@ static const __device__ __constant__ uint8_t ClosestFixedBitsMap[65] = {
  * @return number of values decoded
  */
 template <class T>
-static __device__ uint32_t Integer_RLEv2(
-  orc_bytestream_s* bs, volatile orc_rlev2_state_s* rle, volatile T* vals, uint32_t maxvals, int t)
+static __device__ uint32_t Integer_RLEv2(orc_bytestream_s* bs,
+                                         volatile orc_rlev2_state_s* rle,
+                                         volatile T* vals,
+                                         uint32_t maxvals,
+                                         int t,
+                                         bool has_buffer = false)
```

Comment:
doc for the new parameter is missing.
does `has_buffer` mean "there are buffered values"? Having a hard time understanding the logic added here.

Reply:
has buffer means there are buffered values, and now when it tries to completely fill the buffer there is a chance that number of values that gets decoded here might exceed what buffer can accommodate, so we skip this iteration and process buffered value, and then get back in next iteration where whole buffer is available.

Reply:
Maybe then it should be called has_buffered_values or skip_decode?

Reply:
Updated the doc and also update name of the variable.